### PR TITLE
Feature/default profile names

### PR DESCRIPTION
**Pull request by erik-sn**
This more elegantly handles the case of generating twitter profile names, with any eye towards using it for tinder profile names too.

There are two primary configuration options we're concerned about:

1. `twitter.generateProfileNameFromUsers` - whether or not we automatically generate profile names
2. `twitter.allowEdtiableProfileName` - whether or not we allow players to edit their own profile names

**Various cases:**

- When 1 is true we attempt to generate profile names based on:

1. `firstname_lastname` if both exist
2. only `firstname` if it exists
3. only `lastname` if it exists
4. `phone_number` otherwise

- When 1 is false and 2 is true we show an input field and allow the user to type in a profile name of their choosing
- When 1 is false AND 2 is false we show a select field populated by name (same logic as above) AND phone number:

img

In the future specifying which table/columns to read these values from will make this much more flexible.

### CODE REVIEW

File: phone/src/apps/messages/components/modal/MessageModal.tsx [hunk 1]
```diff
@@ -157,7 +157,7 @@ export const MessageModal = () => {
                 </Button>
               ) : null}
               {!activeMessageGroup?.isGroupChat &&
-              getDisplayByNumber(targetNumber) == targetNumber ? (
+              getDisplayByNumber(targetNumber) === targetNumber ? (
```
Comment (erik-sn) [inline phone/src/apps/messages/components/modal/MessageModal.tsx:160]: random thing my linter highlighted

File: resources/server/players/sv_players.ts [hunk 2]
```diff
@@ -89,6 +89,54 @@ export async function getPlayerInfo(identifier: string): Promise<PlayerQueryResp
   return { firstname, lastname, phone_number };
 }
 
+const clean = (input: string) => (input ? input.replace(/[^0-9a-z]/gi, '') : input);
```
Comment (erik-sn) [inline resources/server/players/sv_players.ts:92]: If anyone has some elite JS to simplify/consolidate/condense all these functions let me know!

File: resources/server/sv_twitter.ts [hunk 3]
```diff
@@ -316,12 +316,41 @@ onNet(events.TWITTER_GET_OR_CREATE_PROFILE, async () => {
...
+      emitNet(events.TWITTER_GET_OR_CREATE_PROFILE_NULL, _source, defaultProfileNames);
+    } else {
+      emitNet(events.TWITTER_GET_OR_CREATE_PROFILE_SUCCESS, _source, profile);
+    }
   } catch (e) {
     twitterLogger.error(`Failed to get or create profile, ${e.message}`, {
       source: _source,
     });
-    emitNet(events.TWITTER_GET_OR_CREATE_PROFILE_FAILURE, _source);
+    emitNet(events.TWITTER_GET_OR_CREATE_PROFILE_FAILURE, _source, null);
```
Comment (TasoOneAsia) [inline resources/server/sv_twitter.ts:333]: If there is a failure is there any reason we have to emit null? Can't it be assumed from a failure event?

File: resources/server/players/sv_players.ts [hunk 4]
```diff
@@ -89,6 +89,54 @@ export async function getPlayerInfo(identifier: string): Promise<PlayerQueryResp
...
+
+  if (firstname && lastname) {
+    return `${firstname}_${lastname}`;
+  } else if (firstname) {
+    return firstname;
+  } else if (lastname) {
+    return lastname;
+  } else if (phone_number) {
+    return phone_number;
+  }
```
Comment (TasoOneAsia) [inline resources/server/players/sv_players.ts:118]: Would be a tiny bit more efficient as a switch but its inconsequential in this case.

**Review by TasoOneAsia — APPROVED** [review]
screenshot

Just some general linter warnings as well but lgtm 👍

**Comment by erik-sn** [comment]
@TasoOneAsia good points, updated with your feedback

**Review by itschip — APPROVED** [review]

**Pull request merged**